bubble-sort.rb

```diff
@@ -0,0 +1,33 @@
+
+def bubble_sort(array)
+  for i in 0..array.size - 2
+    for j in 0..array.size - 2
+      if (array[j] <=> array[j + 1] > 0)
+        tmp = array[j]
+        array[j] = array[j + 1]
+        array[j + 1] = tmp
+      end
+    end
+  end
+
+  return array
+end
+
+def bubble_sort_by(array)
+
+  return nil if !block_given?
```

Comment:
It should not return `nil` when no block is given. `bubble_sort_by` needs to work with and without a block. BTW good work with `spaceship` operator. It's always great to learn newer syntax.

Reply:
OK. Thanks